src/vcpkg/commands.z-applocal.cpp

```diff
@@ -585,17 +597,70 @@ namespace vcpkg
         }
 
         const auto target_installed_bin_dir =
-            real_filesystem.almost_canonical(target_installed_bin_setting->second, VCPKG_LINE_INFO);
-        const auto target_binary_path = real_filesystem.almost_canonical(target_binary->second, VCPKG_LINE_INFO);
+            fs.almost_canonical(target_installed_bin_setting->second, VCPKG_LINE_INFO);
         const auto decoded = decode_from_canonical_bin_dir(target_installed_bin_dir);
-        AppLocalInvocation invocation(real_filesystem,
+
+        // the first binary is special in that it might not be a DLL or might not exist
+        const Path target_binary_path = target_binary->second;
```

Comment:
```suggestion
        const Path& target_binary_path = target_binary->second;
```

Reply:
I think that would be wrong because there is a conversion intended here. (`target_binary->second` is a `std::string`, not a `Path`)

Reply:
In this case I'd use `const Path target_binary_path(target_binary->second)` to make the conversion more obvious.